### ISSUE TITLE
NoClassDefFoundError

### ISSUE THREAD

**Issue by qzsiniong** [vojtechhabarta/typescript-generator#346]
When I use typescript-generator with Spring boot and Spring jpa,I got a NoClassDefFoundError.

```
 ✘  ~/workspace/my-projects/spring-typescript-react  mvn typescript-generator:generate   
[INFO] Scanning for projects...
[INFO] 
[INFO] ------------------< com.qzs:spring-typescript-react >-------------------
[INFO] Building spring-typescript-react 0.1.0
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- typescript-generator-maven-plugin:2.14.505:generate (default-cli) @ spring-typescript-react ---
Running TypeScriptGenerator version 2.14.505
Loading class cz.habarta.typescript.generator.ext.AxiosClientExtension
Configuration: 'AxiosClientExtension' extension set 'generateJaxrsApplicationClient' parameter to 'true'
Configuration: 'AxiosClientExtension' extension set 'restResponseType' parameter to 'Promise<Axios.GenericAxiosResponse<R>>'
Configuration: 'AxiosClientExtension' extension set 'restOptionsType' parameter to '<O>'
Loading Javadoc XML file: /Users/qzs/workspace/my-projects/spring-typescript-react/target/javadoc.xml
Scanning Spring application: com.qzs.Application
[WARNING] Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.BeanDefinitionStoreException: Failed to process import candidates for configuration class [org.springframework.boot.autoconfigure.data.jpa.JpaRepositoriesAutoConfiguration]; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.boot.autoconfigure.data.jpa.JpaRepositoriesAutoConfigureRegistrar]: Unresolvable class definition; nested exception is java.lang.NoClassDefFoundError: org/springframework/data/repository/config/RepositoryConfigurationSource
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 4.281 s
[INFO] Finished at: 2019-05-08T10:22:25+08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal cz.habarta.typescript-generator:typescript-generator-maven-plugin:2.14.505:generate (default-cli) on project spring-typescript-react: Execution default-cli of goal cz.habarta.typescript-generator:typescript-generator-maven-plugin:2.14.505:generate failed: Failed to process import candidates for configuration class [org.springframework.boot.autoconfigure.data.jpa.JpaRepositoriesAutoConfiguration]; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [org.springframework.boot.autoconfigure.data.jpa.JpaRepositoriesAutoConfigureRegistrar]: Unresolvable class definition; nested exception is java.lang.NoClassDefFoundError: org/springframework/data/repository/config/RepositoryConfigurationSource: org.springframework.data.repository.config.RepositoryConfigurationSource -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
```

and my pom.xml like this
```xml
<?xml version="1.0" encoding="UTF-8" ?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <groupId>com.qzs</groupId>
    <artifactId>spring-typescript-react</artifactId>
    <version>0.1.0</version>
    <properties>
        <java.version>1.8</java.version>
        <typescript-generator.version>2.14.505</typescript-generator.version>
        <kaptcha.version>0.0.9</kaptcha.version>
    </properties>
    <parent>
        <groupId>org.springframework.boot</groupId>
        <artifactId>spring-boot-starter-parent</artifactId>
        <version>2.0.5.RELEASE</version>
    </parent>
    <dependencies>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-web</artifactId>
        </dependency>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-actuator</artifactId>
        </dependency>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-data-jpa</artifactId>
        </dependency>
        <dependency>
            <groupId>mysql</groupId>
            <artifactId>mysql-connector-java</artifactId>
        </dependency>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-starter-test</artifactId>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.springframework.boot</groupId>
            <artifactId>spring-boot-devtools</artifactId>
            <optional>true</optional>
        </dependency>
        <dependency>
            <groupId>com.spring4all</groupId>
            <artifactId>swagger-spring-boot-starter</artifactId>
            <version>1.7.1.RELEASE</version>
        </dependency>
        <dependency>
            <groupId>com.github.caspar-chen</groupId>
            <artifactId>swagger-ui-layer</artifactId>
            <version>1.1.3</version>
        </dependency>
        <dependency>
            <groupId>org.apache.shiro</groupId>
            <artifactId>shiro-spring-boot-web-starter</artifactId>
            <version>1.4.0</version>
        </dependency>
        <dependency>
            <groupId>com.github.axet</groupId>
            <artifactId>kaptcha</artifactId>
            <version>${kaptcha.version}</version>
        </dependency>
        <!-- https://mvnrepository.com/artifact/com.alibaba/fastjson -->
        <dependency>
            <groupId>com.alibaba</groupId>
            <artifactId>fastjson</artifactId>
            <version>1.2.57</version>
        </dependency>
        <!-- https://mvnrepository.com/artifact/org.projectlombok/lombok -->
        <dependency>
            <groupId>org.projectlombok</groupId>
            <artifactId>lombok</artifactId>
            <version>1.18.6</version>
            <scope>provided</scope>
        </dependency>
    </dependencies>
    <build>
        <plugins>
            <plugin>
                <groupId>org.springframework.boot</groupId>
                <artifactId>spring-boot-maven-plugin</artifactId>
            </plugin>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-javadoc-plugin</artifactId>
                <version>2.10.3</version>
                <executions>
                    <execution>
                        <id>xml-doclet</id>
                        <phase>process-classes</phase>
                        <goals>
                            <goal>javadoc</goal>
                        </goals>
                        <configuration>
                            <doclet>com.github.markusbernhardt.xmldoclet.XmlDoclet</doclet>
                            <additionalparam>-d ${project.build.directory}</additionalparam>
                            <useStandardDocletOptions>false</useStandardDocletOptions>
                            <docletArtifact>
                                <groupId>com.github.markusbernhardt</groupId>
                                <artifactId>xml-doclet</artifactId>
                                <version>1.0.5</version>
                            </docletArtifact>
                        </configuration>
                    </execution>
                </executions>
            </plugin>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-compiler-plugin</artifactId>
                <version>3.7.0</version>
                <configuration>
                    <source>1.8</source>
                    <target>1.8</target>
                    <compilerArgs>
                        <arg>-parameters</arg>
                    </compilerArgs>
                </configuration>
            </plugin>
            <plugin>
                <groupId>cz.habarta.typescript-generator</groupId>
                <artifactId>typescript-generator-maven-plugin</artifactId>
                <version>${typescript-generator.version}</version>
                <executions>
                    <execution>
                        <id>generate</id>
                        <goals>
                            <goal>generate</goal>
                        </goals>
                        <phase>process-classes</phase>
                    </execution>
                </executions>
                <configuration>
                    <jsonLibrary>jackson2</jsonLibrary>
                    <outputFileType>implementationFile</outputFileType>
                    <outputKind>module</outputKind>
                    <classes>
                        <class>com.qzs.Application</class>
                    </classes>
                    <javadocXmlFiles>
                        <file>target/javadoc.xml</file>
                    </javadocXmlFiles>
                    <scanSpringApplication>true</scanSpringApplication>
                    <generateSpringApplicationClient>true</generateSpringApplicationClient>
                    <optionalProperties>useLibraryDefinition</optionalProperties>
                    <extensions>
                        <extension>cz.habarta.typescript.generator.ext.AxiosClientExtension</extension>
                    </extensions>
                    <outputFile>./webapp/src/api.ts</outputFile>
                    <noTslintDisable>false</noTslintDisable>
                    <noEslintDisable>false</noEslintDisable>
                    <addTypeNamePrefix>I</addTypeNamePrefix>
                </configuration>
                <dependencies>
                    <dependency>
                        <groupId>cz.habarta.typescript-generator</groupId>
                        <artifactId>typescript-generator-spring</artifactId>
                        <version>${typescript-generator.version}</version>
                    </dependency>
                </dependencies>
            </plugin>
        </plugins>
    </build>
</project>
```

**Comment by vojtechhabarta** [comment]
I don't know what exactly causes this problem but it is probably some incompatibility between versions of Spring artifacts used by typescript-generator and your application.

So instead of using `scanSpringApplication` parameter (which loads your Spring application and causes the problem) you can use for example `classPatterns` parameter to specify your controller classes.

**Comment by qzsiniong** [comment]
@vojtechhabarta thanks, it's works.

**Comment by RobbinBaauw** [comment]
Could it be a matter of upgrading to a newer version? I see `2.1.1.RELEASE` is used, which is almost a year old now. I currently have errors which would probably be fixed by an upgrade to `2.1.9.RELEASE` (google searches on the error said the errors were fixed on `2.1.3`)

**Comment by RobbinBaauw** [comment]
Hm unfortunately that didn't do it, I get a different error though: `Attribute 'proxyBeanMethods' in annotation [org.springframework.boot.actuate.autoconfigure.web.ManagementContextConfiguration] is declared as an @AliasFor nonexistent attribute 'proxyBeanMethods' in annotation [org.springframework.context.annotation.Configuration].`

I will try and look for the source
